src/java/org/apache/cassandra/cql3/restrictions/ClusteringElements.java

```diff
@@ -470,4 +424,80 @@ public ClusteringBound<?> toBound(boolean isStart, boolean isInclusive)
             return isEmpty() ? BufferClusteringBound.TOP : super.toBound(isStart, isInclusive);
         }
     }
+
+    public static class ClusteringElementsComparator implements Comparator<ClusteringElements>
```

Comment:
I would add some Java doc for this comparator and mostly for the `forCQL`.  It should be clear without checking the usages.